roachtest: cdc/initial-scan-only/parquet failed

roachtest.cdc/initial-scan-only/parquet [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/11980154?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/11980154?buildTab=artifacts#/cdc/initial-scan-only/parquet) on master @ [27e7c166b5591b0e05bb2fd23780309b490d7af9](https://github.com/cockroachdb/cockroach/commits/27e7c166b5591b0e05bb2fd23780309b490d7af9):


```
(latency_verifier.go:185).assertValid: initial scan did not complete
(monitor.go:153).Wait: monitor failure: monitor user task failed: t.Fatal() was called
test artifacts and logs in: /artifacts/cdc/initial-scan-only/parquet/run_1
```
<p>Parameters: <code>ROACHTEST_arch=amd64</code>
, <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=16</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_fs=ext4</code>
, <code>ROACHTEST_localSSD=true</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-11980154/cdc-initial-scan-only-parquet/1695983545461/1695984352086)

</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*cdc/initial-scan-only/parquet.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31932

I just ran the non-parquet test and it failed. I commented out the stats collection bit, but I don't think that matters..
```
exportStatsFile := ct.startStatsCollection()
exportStatsFile()
```

```
changefeed: 23:28:35 latency_verifier.go:187: running highwater: 1970-01-01 00:00:00 +0000 UTC
changefeed: 23:28:36 latency_verifier.go:187: running highwater: 1970-01-01 00:00:00 +0000 UTC
changefeed: 23:28:37 latency_verifier.go:187: running highwater: 1970-01-01 00:00:00 +0000 UTC
changefeed: 23:28:38 cdc.go:1050: parquet test completed: 436783000
23:28:38 cdc.go:342: test status: cdcTester closing
changefeed: 23:28:38 latency_verifier.go:164: poller stopped at 437580000
changefeed: 23:28:38 cdc.go:523: latency verifier asserting validity 437652000
23:28:38 test_impl.go:357: test failure #1: full stack retained in failure_1.log: (latency_verifier.go:199).assertValid: initial scan did not complete
23:28:38 test_impl.go:357: test failure #2: full stack retained in failure_2.log: (monitor.go:153).Wait: monitor failure: monitor user task failed: t.Fatal() was called
23:28:38 test_runner.go:1146: test completed with failure(s)
23:28:38 test_runner.go:1177: skipping post test assertions as test failed
23:28:38 test_runner.go:1180: running test teardown (test-teardown.log)
23:30:28 test_runner.go:982: metrics: https://go.crdb.dev/roachtest-grafana/jayants-1696807117/cdc-initial-scan-only/1696807436087/1696807931929
23:30:28 test_runner.go:1024: [w0] --- FAIL: cdc/initial-scan-only (339.27s)
(latency_verifier.go:199).assertValid: initial scan did not complete
(monitor.go:153).Wait: monitor failure: monitor user task failed: t.Fatal() was called
test artifacts and logs in: artifacts/cdc/initial-scan-only/run_1
--- FAIL: cdc/initial-scan-only (339.27s)
(latency_verifier.go:199).assertValid: initial scan did not complete
```

--- a/pkg/ccl/changefeedccl/changefeed_processors.go
+++ b/pkg/ccl/changefeedccl/changefeed_processors.go
@@ -654,9 +654,6 @@ func (ca *changeAggregator) Next() (rowenc.EncDatumRow, *execinfrapb.ProducerMet
 				// If the poller errored first, that's the
 				// interesting one, so overwrite `err`.
 				if kvFeedErr := ca.checkKVFeedErr(); kvFeedErr != nil {
-					if log.V(1) {
-						log.Infof(ca.Ctx(), "overwriting error %s", err)
-					}
 					err = kvFeedErr
 				}
 			}
@@ -1389,9 +1386,6 @@ func (cf *changeFrontier) forwardFrontier(resolved jobspb.ResolvedSpan) error {
 		return err
 	}
 
-	if log.V(1) {
-		log.Infof(cf.Ctx(), "forwarding frontier. changed: %t, resolved: %s", frontierChanged, resolved)
-	}
 	cf.maybeLogBehindSpan(frontierChanged)
 
 	// If frontier changed, we emit resolved timestamp.
@@ -1478,21 +1472,12 @@ func (cf *changeFrontier) maybeCheckpointJob(
 		checkpoint.Spans, checkpoint.Timestamp = cf.frontier.getCheckpointSpans(maxBytes)
 	}
 
-	if log.V(1) {
-		log.Infof(cf.Ctx(), "maybe checkpointing job. updateCheckpoint: %t, updateHighWater: %t",
-			updateCheckpoint, updateHighWater)
-	}
-
 	if updateCheckpoint || updateHighWater {
 		if cf.knobs.ShouldCheckpointToJobRecord != nil && !cf.knobs.ShouldCheckpointToJobRecord(cf.frontier.Frontier()) {
 			return false, nil
 		}
 		checkpointStart := timeutil.Now()
 		updated, err := cf.checkpointJobProgress(cf.frontier.Frontier(), checkpoint)
-		if log.V(1) {
-			log.Infof(cf.Ctx(), "checkpointed job progress: updated: %t, highwater: %s, err: %v",
-				updated, cf.frontier.Frontier(), err)
-		}
 		if err != nil {
 			return false, err
 		}

--- a/pkg/cmd/roachtest/tests/cdc.go
+++ b/pkg/cmd/roachtest/tests/cdc.go
@@ -1088,11 +1088,6 @@ func registerCDC(r registry.Registry) {
 		CompatibleClouds: registry.AllExceptAWS,
 		Suites:           registry.Suites(registry.Nightly),
 		Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
-			vmoduleLevel = 1
-			defer func() {
-				vmoduleLevel = 0
-			}()
-
 			ct := newCDCTester(ctx, t, c)
 			defer ct.Close()
 

--- a/pkg/cmd/roachtest/tests/cdc.go
+++ b/pkg/cmd/roachtest/tests/cdc.go
@@ -495,7 +495,15 @@ func (ct *cdcTester) newChangefeed(args feedArgs) changefeedJob {
 	return cj
 }
 
-func (ct *cdcTester) runFeedLatencyVerifier(cj changefeedJob, targets latencyTargets) {
+// runFeedLatencyVerifier runs a goroutine which polls the various latencies
+// for a changefeed job (initial scan latency, etc) and asserts that they
+// are below the specified targets.
+//
+// It returns a function which blocks until the job succeeds and verification
+// on the succeeded job completes.
+func (ct *cdcTester) runFeedLatencyVerifier(
+	cj changefeedJob, targets latencyTargets,
+) (waitForCompletion func()) {
 	info, err := getChangefeedInfo(ct.DB(), cj.jobID)
 	if err != nil {
 		ct.t.Fatalf("failed to get changefeed info: %s", err.Error())
@@ -512,8 +520,12 @@ func (ct *cdcTester) runFeedLatencyVerifier(cj changefeedJob, targets latencyTar
 	)
 	verifier.statementTime = info.statementTime
 
+	finished := make(chan struct{})
 	ct.mon.Go(func(ctx context.Context) error {
-		err := verifier.pollLatency(ctx, ct.DB(), cj.jobID, time.Second, ct.doneCh)
+		defer func() {
+			finished <- struct{}{}
+		}()
+		err := verifier.pollLatencyUntilJobSucceeds(ctx, ct.DB(), cj.jobID, time.Second, ct.doneCh)
 		if err != nil {
 			return err
 		}
@@ -522,6 +534,13 @@ func (ct *cdcTester) runFeedLatencyVerifier(cj changefeedJob, targets latencyTar
 		verifier.maybeLogLatencyHist()
 		return nil
 	})
+
+	return func() {
+		select {
+		case <-ct.ctx.Done():
+		case <-finished:
+		}
+	}
 }
 
 func (cj *changefeedJob) runFeedPoller(
@@ -1025,10 +1044,11 @@ func registerCDC(r registry.Registry) {
 				targets:  allTpccTargets,
 				opts:     map[string]string{"initial_scan": "'only'"},
 			})
-			ct.runFeedLatencyVerifier(feed, latencyTargets{
+			waitForCompletion := ct.runFeedLatencyVerifier(feed, latencyTargets{
 				initialScanLatency: 30 * time.Minute,
 			})
-			feed.waitForCompletion()
+			waitForCompletion()
+
 			exportStatsFile()
 		},
 	})
@@ -1083,10 +1103,11 @@ func registerCDC(r registry.Registry) {
 				targets:  allTpccTargets,
 				opts:     map[string]string{"initial_scan": "'only'", "format": "'parquet'"},
 			})
-			ct.runFeedLatencyVerifier(feed, latencyTargets{
+			waitForCompletion := ct.runFeedLatencyVerifier(feed, latencyTargets{
 				initialScanLatency: 30 * time.Minute,
 			})
-			feed.waitForCompletion()
+			waitForCompletion()
+
 		},
 	})
 	r.Add(registry.TestSpec{

--- a/pkg/cmd/roachtest/tests/cdc_bench.go
+++ b/pkg/cmd/roachtest/tests/cdc_bench.go
@@ -166,8 +166,6 @@ func formatSI(num int64) string {
 	return fmt.Sprintf("%d%s", int64(numSI), suffix)
 }
 
-var vmoduleLevel = 0
-
 // makeCDCBenchOptions creates common cluster options for CDC benchmarks.
 func makeCDCBenchOptions() (option.StartOpts, install.ClusterSettings) {
 	opts := option.DefaultStartOpts()
@@ -222,10 +220,6 @@ func makeCDCBenchOptions() (option.StartOpts, install.ClusterSettings) {
 	// due to elevated goroutine scheduling latency.
 	// Current default is 4s which should be sufficient.
 	// settings.Env = append(settings.Env, "COCKROACH_NETWORK_TIMEOUT=6s")
-
-	if vmoduleLevel > 0 {
-		opts.RoachprodOpts.ExtraArgs = []string{"--vmodule=changefeed_processors=2"}
-	}
 
 	return opts, settings
 }

--- a/pkg/cmd/roachtest/tests/cluster_to_cluster.go
+++ b/pkg/cmd/roachtest/tests/cluster_to_cluster.go
@@ -930,7 +930,7 @@ func (rd *replicationDriver) main(ctx context.Context) {
 
 	latencyMonitor := rd.newMonitor(ctx)
 	latencyMonitor.Go(func(ctx context.Context) error {
-		return lv.pollLatency(ctx, rd.setup.dst.db, ingestionJobID, time.Second, workloadDoneCh)
+		return lv.pollLatencyUntilJobSucceeds(ctx, rd.setup.dst.db, ingestionJobID, time.Second, workloadDoneCh)
 	})
 	defer latencyMonitor.Wait()
 

--- a/pkg/cmd/roachtest/tests/latency_verifier.go
+++ b/pkg/cmd/roachtest/tests/latency_verifier.go
@@ -148,7 +148,9 @@ func (lv *latencyVerifier) noteHighwater(highwaterTime time.Time) {
 	}
 }
 
-func (lv *latencyVerifier) pollLatency(
+// pollLatencyUntilJobSucceeds polls the changefeed latency until it is
+// signalled to stop or the job completes.
+func (lv *latencyVerifier) pollLatencyUntilJobSucceeds(
 	ctx context.Context, db *gosql.DB, jobID int, interval time.Duration, stopper chan struct{},
 ) error {
 	for {
@@ -171,6 +173,8 @@ func (lv *latencyVerifier) pollLatency(
 		status := info.GetStatus()
 		if status == "succeeded" {
 			lv.noteHighwater(info.GetFinishedTime())
+			lv.logger.Printf("latency poller shutting down due to changefeed completion")
+			return nil
 		} else if status == "running" {
 			lv.noteHighwater(info.GetHighWater())
 		} else {

